include/zephyr/bluetooth/gatt.h

```diff
@@ -1558,6 +1558,7 @@ struct bt_gatt_discover_params;
  *  @param conn Connection object.
  *  @param attr Attribute found, or NULL if not found.
  *  @param params Discovery parameters given.
+ *  @param err 0 on success, -ECONNRESET on connection lost, -EINVAL on invalid data
```

Comment:
I would prefer to reserve `EINVAL` for API violations. Errors caused by the remote should be something different. `EPROTO` makes sense in this case.